### PR TITLE
refactor(@desktop/chat): use StatusQ components for chat user list

### PR DESCRIPTION
**Pull request by PascalPrecht**
This replaces the self-built components for the user list with StatusQ's
list item component and also addresses some of the quirks that existed in the UI
related to that original component.

The user list is now responsive and properly aligned.

‼️ **This depends on StatusQ v0.18.0** ‼️

### CODE REVIEW

**Review by alexandraB99 — CHANGES_REQUESTED** [review]
Great work! just one detail I noticed is that in minimized state the name text is still there:
chatMembers
commMembers

**Comment by PascalPrecht** [comment]
Nice!! Good catch. Will update

File: ui/app/AppLayouts/Chat/controls/UserDelegate.qml [hunk 1]
```diff
@@ -9,138 +9,95 @@ import StatusQ.Components 0.1
...
+        height: 15
+        anchors.left: parent.left
+        anchors.leftMargin: 22
+        anchors.bottom: parent.bottom
+        anchors.bottomMargin: 6
+        visible: root.isOnline && !((root.statusType === -1) && (lastSeenMinutesAgo > 7))
+        border.width: 3
+        border.color: root.sensor.containsMouse ? Theme.palette.baseColor2 : Theme.palette.baseColor4
+        property real lastSeenMinutesAgo: ((currentTime/1000 - parseInt(lastSeen)) / 60);
+        color: {
```
Comment (Khushboo-dev-cpp) [inline ui/app/AppLayouts/Chat/controls/UserDelegate.qml:87]: just a suggestion maybe to move this to the store? the logic of calculating the color of the status indicator?

**Review by Khushboo-dev-cpp — APPROVED** [review]

File: ui/app/AppLayouts/Chat/controls/UserDelegate.qml (hunk 1 above)
Comment (PascalPrecht) [reply ui/app/AppLayouts/Chat/controls/UserDelegate.qml:87]: That's a good suggestion. We could even go as far as adding an enum to this component and the variant needs to be set at the call side (using that store API)

Comment (Khushboo-dev-cpp) [reply ui/app/AppLayouts/Chat/controls/UserDelegate.qml:87]: aaah yes, but then we will again be putting if cases against the ENUM in the qml? 
Also shouldn't this simply come from backend? at least the logic of calculating the various states?

**Comment by PascalPrecht** [comment]
Quick update here: I can't reproduce @alexandraB99 case, so waiting for here to reconfirm

**Review by alexandraB99 — APPROVED** [review]
can't reproduce either, probably I didn't update or I don't know. Thanks @PascalPrecht, looks good!

**Comment by PascalPrecht** [comment]
Sweet thanks!!

**Pull request merged**